pkg/chunk/cassandra/storage_client.go

```diff
@@ -121,6 +121,27 @@ func (cfg *Config) createKeyspace() error {
 	return errors.WithStack(err)
 }
 
+// PasswordAuthenticator creates the gocql Authenticator object
+type PasswordAuthenticator struct {
+	Username string
+	Password string
+}
+
+// Challenge implements the Authenticator interface for gocql
+func (p PasswordAuthenticator) Challenge(req []byte) ([]byte, gocql.Authenticator, error) {
+	resp := make([]byte, 2+len(p.Username)+len(p.Password))
+	resp[0] = 0
+	copy(resp[1:], p.Username)
+	resp[len(p.Username)+1] = 0
+	copy(resp[2+len(p.Username):], p.Password)
+	return resp, nil, nil
+}
+
+// Success implements the Authenticator interface for gqcql
+func (p PasswordAuthenticator) Success(data []byte) error {
+	return nil
```

Comment:
I know nothing about Cassandra, so maybe this is obvious, but I found this line surprising.
We always return no error without checking anything?